SqlInstall.ps1

```diff
@@ -0,0 +1,556 @@
+SECURITYMODE="SQL"
+; Default directory for the Database Engine log files. 
+SQLUSERDBLOGDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"
+; Default directory for the Database Engine backup files. 
+SQLBACKUPDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Backup"
+; Default directory for the Database Engine user databases. 
+SQLUSERDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"
+; Directory for Database Engine TempDB files. 
+SQLTEMPDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"
+; Provision current user as a Database Engine system administrator for %SQL_PRODUCT_SHORT_NAME% Express. 
+ADDCURRENTUSERASSQLADMIN="True"
+; Specify 0 to disable or 1 to enable the TCP/IP protocol. 
+TCPENABLED="0"
+; Specify 0 to disable or 1 to enable the Named Pipes protocol. 
+NPENABLED="0"
+; Startup type for Browser Service. 
+BROWSERSVCSTARTUPTYPE="Automatic"
+'@
+
+    # Check if there are designated drives for Data, Log, TempDB, Back-up and Application.
+    If ($DataVolume -eq $null -or $DataVolume -eq '') {
+        $DataVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne  $_.DriveLetter -and $_.FileSystemLabel -like '*Data*'} | 
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($LogVolume -eq $null -or $LogVolume -eq '') {
+        $LogVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*Log*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($TempVolume -eq $null -or $TempVolume -eq '') {
+        $TempVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*TempDB*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($AppVolume -eq $null -or $AppVolume -eq '') {
+        $AppVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*App*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    if ($BackupVolume -eq $null -or $BackupVolume -eq '') {
+        $BackupVolume = Get-Volume | 
+            Where-Object {$_.DriveType -EQ 'Fixed' -and $null -ne $_.DriveLetter -and $_.FileSystemLabel -like '*Backup*'} |  
+            Select-Object -ExpandProperty DriveLetter
+    }
+    #Check the number of cores available on the server. Summed because every processor can contain multiple cores
+    $NumberOfCores = Get-WmiObject -Class Win32_processor |  
+        Measure-Object NumberOfLogicalProcessors -Sum | 
+        Select-Object -ExpandProperty sum
+
+    IF ($NumberOfCores -gt 8)
+    { $NumberOfCores = 8 }
+
+    #Get the amount of available memory. If it's more than 40 GB, give the server 10% of the memory, else reserve 4 GB.
+
+    $ServerMemory = Get-WmiObject -Class win32_physicalmemory | 
```

Comment:
Get-WmiObject is slower than the new Get-CimInstance, but in our toolset, we use `Get-DbaCmObject` which has the same syntax. Nevertheless, we do have a command that calculates this that you'd want to take advantage of after the install is perofrmed - `Test-DbaMaxMemory`